sunpycube/cube/datacube.py

```diff
@@ -213,6 +257,56 @@ def _choose_x_slice(self, offset):
 
         return arr
 
+    @classmethod
+    def _new_instance(cls, data, wcs, errors=None, **kwargs):
+        """
+        Instantiate a new instance of this class using given data.
+        """
+        return cls(data, wcs, errors=errors, **kwargs)
+
+    def truncate(self, slice_data):
+        """
+        Truncates the data on any dimension.
+
+        Parameters
+        ----------
+        slice_data: `tuple` or `list` or `slice`
+            Given a list/tuple/slice like slice_data=(slice(2,3), None, slice(0,6)).
+            This will return cube with data sliced in 1st dimension between 2 and 3,
+            2nd dimension returned as it is and 3rd dimension sliced between 0 and 6.
+
+        Examples
+        --------
+        >>> from sunpycube.cube.datacube import Cube
+        >>> file = #location in computer
+        >>> cube = Cube(data,wcs)
+        >>> new_cube = cube.truncate((slice(3,4), None, slice(0,5)))
+        >>> new_cube1 = cube.truncate([slice(3,4), None, slice(0,5)])
+        >>> new_cube2 = cube.truncate(slice(3,4))
+
+
+        """
+        data = self.data
+        wcs = self.axes_wcs
+        mask = self.mask
+
+        if isinstance(slice_data, slice):
+            data = data[slice_data]
+            wcs = wcs.slice(slice_data, numpy_order=False)
+            mask = mask[slice_data]
+            return self._new_instance(data, wcs, mask=mask)
+
+        slice_parameters_list = list(slice_data)
+        for i, _slice in enumerate(slice_parameters_list):
+            if _slice is None:
+                slice_parameters_list[i] = slice(0, data.shape[i])
+
+        data = data[slice_parameters_list]
+        wcs = wcs.slice(slice_parameters_list, numpy_order=False)
```

Comment:
```slice_parameters_list``` must be a tuple when inputting it to ```wcs.slice```.

Also, delete ```numpy_order=False```.  The default is ```True``` which is I'm pretty sure now is actually what we want.